src/components/Context/injectEnvironment.js

```diff
@@ -10,18 +10,28 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import { deepAssign } from "../../utils";
+import { deepAssign, isNumber } from "../../utils";
 
 export default window => {
   return xdm => {
-    const { innerWidth, innerHeight } = window;
+    const {
+      document: { documentElement: { clientWidth, clientHeight } = {} }
+    } = window;
+
     const environment = {
-      type: "browser",
-      browserDetails: {
-        viewportWidth: innerWidth,
-        viewportHeight: innerHeight
-      }
+      type: "browser"
     };
+    const hasWidth = isNumber(clientWidth) && clientWidth >= 0;
+    const hasHeight = isNumber(clientHeight) && clientHeight >= 0;
+    if (hasWidth || hasHeight) {
```

Comment:
Is there a valid use-case for not doing && here?

Reply:
I can't think of one. I like it.